### PR TITLE
Enhance logs

### PR DESCRIPTION
**Pull request by valaparthvi**
kind/enhancement
**What this PR does / why we need it**:

1. Use [cluster-name (id: cluster-id)] instead of either [cluster-name] or [cluster-id]
2. Log more information

**Which issue(s) this PR fixes**
Issue #

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits into logical changes
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests
- [ ] backport needed

### CODE REVIEW

File: controller/aks-cluster-config-handler.go [hunk 1]
```diff
@@ -728,7 +729,8 @@ func (h *Handler) updateUpstreamClusterState(ctx context.Context, config *aksv1.
 	// check tags for update
 	if config.Spec.Tags != nil {
 		if !reflect.DeepEqual(config.Spec.Tags, upstreamSpec.Tags) {
-			logrus.Infof("Updating tags for cluster [%s]", config.Spec.ClusterName)
+			logrus.Infof("Updating tags to %v for cluster [%s (id: %s)]", config.Spec.Tags, config.Spec.ClusterName, config.Name)
```
Comment (mjura) [inline controller/aks-cluster-config-handler.go:732]:
```suggestion
			logrus.Infof("Updating tags for cluster [%s (id: %s)]", config.Spec.ClusterName, config.Name)
```

File: controller/aks-cluster-config-handler.go [hunk 2]
```diff
@@ -762,7 +764,8 @@ func (h *Handler) updateUpstreamClusterState(ctx context.Context, config *aksv1.
 	// check Kubernetes version for update
 	if config.Spec.KubernetesVersion != nil {
 		if aks.String(config.Spec.KubernetesVersion) != aks.String(upstreamSpec.KubernetesVersion) {
-			logrus.Infof("Updating kubernetes version for cluster [%s]", config.Spec.ClusterName)
+			logrus.Infof("Updating kubernetes version to %s for cluster [%s (id: %s)]", aks.String(config.Spec.KubernetesVersion), config.Spec.ClusterName, config.Name)
```
Comment (mjura) [inline controller/aks-cluster-config-handler.go:767]:
```suggestion
			logrus.Infof("Updating kubernetes version for cluster [%s (id: %s)]", config.Spec.ClusterName, config.Name)
```

**Review by mjura — CHANGES_REQUESTED** [review]
there are still some places for improvements

File: controller/aks-cluster-config-handler.go (hunk 2 above)
Comment (valaparthvi) [reply controller/aks-cluster-config-handler.go:767]: Why are you suggesting a revert? Same question for the above suggestion.

**Review by furkatgofurov7 — APPROVED** [review]

**Review by Danil-Grigorev — APPROVED** [review]

**Comment by valaparthvi** [comment]
@mjura If you are okay with the changes, can we merge this?

**Review by mjura — APPROVED** [review]
LGTM

**Pull request merged**